internal/vrf/retrieve.go

```diff
@@ -0,0 +1,67 @@
+package vrf
+
+import (
+	"context"
+	"fmt"
+
+	metal "github.com/equinix-labs/metal-go/metal/v1"
+	"github.com/spf13/cobra"
+)
+
+func (c *Client) Retrieve() *cobra.Command {
+	var projectID, metro, vrfID string
+
+	// retrieveVrfsCmd represents the retrieveMetalGateways command
+	retrieveVrfsCmd := &cobra.Command{
+		Use:     `get -p <project_UUID> [-i <vrf_id>]`,
+		Aliases: []string{"list"},
+		Short:   "Lists VRFs.",
+		Long:    "Retrieves a list of all VRFs for the specified project or the details of the specified VRF ID. Either a project ID or a VRF ID is required.",
+		Example: ` # Gets the details of the specified device
+  metal vrf get -i 3b0795ba-ec9a-4a9e-83a7-043e7e11407c
+
+  # Lists VRFs for project 3b0795ba-ec9a-4a9e-83a7-043e7e11407c:
+  metal vrf list -p 3b0795ba-ec9a-4a9e-83a7-043e7e11407c`,
+
+		RunE: func(cmd *cobra.Command, args []string) error {
+			cmd.SilenceUsage = true
+			inc := []string{}
+			exc := []string{}
+			var vrfsList *metal.VrfList
+			var vrf *metal.Vrf
+			var vrfs []metal.Vrf
+			var err error
+			if vrfID != "" {
+				vrf, _, err = c.Service.FindVrfById(context.Background(), vrfID).Include(c.Servicer.Includes(inc)).Exclude(c.Servicer.Excludes(exc)).Execute()
+				if err != nil {
+					return fmt.Errorf("error when calling `VRFsApi.FindVrfById``: %w", err)
+				}
+				vrfs[0] = *vrf
+			} else {
+				vrfsList, _, err = c.Service.FindVrfs(context.Background(), projectID).Metro(metro).Include(c.Servicer.Includes(inc)).Exclude(c.Servicer.Excludes(exc)).Execute() //c.Service.FindVrfs(context.Background(), projectID).Include(c.Servicer.Includes(inc)).Exclude(c.Servicer.Excludes(exc)).Metro(metro).Execute()
+				if err != nil {
+					return fmt.Errorf("error when calling `VRFsApi.FindVrfs``: %w", err)
+				}
+				vrfs = vrfsList.GetVrfs()
+			}
+
+			data := make([][]string, len(vrfs))
+
+			for i, vrf := range vrfs {
+				var IPRanges string
+				for _, ipranges := range vrf.GetIpRanges() {
+					IPRanges += ipranges
```

Comment:
`IPRanges += ipranges` This will simply concatenate 2 IP ranges without any delimiter and wont make any sense. Use `strings.Join(vrf. GetIpRanges(), "\n")`